homeassistant/components/binary_sensor/template.py

```diff
@@ -39,6 +41,8 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
                           device)
             continue
 
+        entity_ids = device_config.get(ATTR_ENTITY_ID)
```

Comment:
`track_state_change` allows passing in the `MATCH_ALL` as `entity_id`. So you can do:

```python
from homeassistant.const import MATCH_ALL

entity_ids = device_config.get(ATTR_ENTITY_ID, MATCH_ALL)
```
_(move it below value_template in this file btw, makes no sense to make it it's own code block)_

Once this is done, you can remove all logic to listen to state change event and always use `track_state_change`